pkg/cable/wireguard/driver.go

```diff
@@ -0,0 +1,411 @@
+		}
+	}
+
+	klog.V(log.DEBUG).Infof("Done removing endpoint for cluster %s", remoteEndpoint.Spec.ClusterID)
+	return nil
+}
+
+func (w *wireguard) GetActiveConnections(clusterID string) ([]string, error) {
+	// force caller to skip duplicate handling
+	return make([]string, 0), nil
+}
+
+// Create new wg link and assign addr from local subnets
+func setWGLink(w *wireguard, localSubnets []string) error {
+
+	// delete existing wg device if needed
+	if link, err := netlink.LinkByName(DefaultDeviceName); err == nil {
+		// delete existing device
+		if err := netlink.LinkDel(link); err != nil {
+			return fmt.Errorf("failed to delete existing WireGuard device: %v", err)
+		}
+	}
+
+	// create the wg device (ip link add dev $DefaultDeviceName type wireguard)
+	la := netlink.NewLinkAttrs()
+	la.Name = DefaultDeviceName
+	link := &netlink.GenericLink{
+		LinkAttrs: la,
+		LinkType:  "wireguard",
+	}
+	if err := netlink.LinkAdd(link); err == nil {
+		w.link = link
+	} else {
+		return fmt.Errorf("failed to add WireGuard device: %v", err)
+	}
+
+	// parse localSubnets and get internal address
+	w.localSubnets = parseSubnets(localSubnets)
+	ip, err := discoverInternalIP(w.localSubnets)
+	if err != nil {
+		klog.V(log.DEBUG).Infof("Using endpoint IP as internal address; %v", err)
+		ip = endpointIP(&w.localEndpoint)
+	}
+	klog.V(log.DEBUG).Infof("Setting interface address to  %s", ip)
+
+	// setup local address (ip address add dev $DefaultDeviceName $PublicIP
+	localIP, err := netlink.ParseAddr(ip + "/32")
+	if err != nil {
+		// try again as CIDR
+		if localIP, err = netlink.ParseAddr(ip); err != nil {
+			return fmt.Errorf("failed to parse IP address %s: %v", ip, err)
+		}
+	}
+	if err = netlink.AddrAdd(w.link, localIP); err != nil {
+		return fmt.Errorf("failed to add local address: %v", err)
+	}
+
+	return nil
+}
+
+// find internal host IP inside one of the local CIDRs
+// TODO move this to utils
+func discoverInternalIP(cidrs []net.IPNet) (string, error) {
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return "", fmt.Errorf("net.InterfaceAddrs() returned error : %v", err)
+	}
+
+	for _, a := range addrs {
+		ip, _, err := net.ParseCIDR(a.String())
+		if err != nil {
+			klog.V(log.DEBUG).Infof("Skipping local address %v, unable to ParseCIDR: %v", a, err)
+			continue
+		}
+		if ip.To4() == nil {
+			klog.V(log.DEBUG).Infof("Skipping local address %+v: not IP4", ip)
+			continue
+		}
+		for _, c := range cidrs {
+			if c.Contains(ip) {
+				return ip.String(), nil
+			}
+		}
+	}
+	return "", fmt.Errorf("could not find internal address in local subnets")
+}
+
+// parse CIDR string and skip errors
+func parseSubnets(subnets []string) []net.IPNet {
+	nets := make([]net.IPNet, 0, len(subnets))
+	for _, sn := range subnets {
+		_, cidr, err := net.ParseCIDR(sn)
+		if err != nil {
+			// this should not happen. Log and continue
+			klog.Errorf("failed to parse subnet %s: %v", sn, err)
```

Comment:
Should continue here and not fall thru to L367.

Reply:
oops. good catch. Done.